hazardous/metrics/brier_score.py

```diff
@@ -0,0 +1,471 @@
+import warnings
+
+import numpy as np
+from sklearn.utils.validation import check_random_state
+
+from .._ipcw import IpcwEstimator
+from ..utils import check_event_of_interest, check_y_mean_increasing, check_y_survival
+
+
+class BrierScoreComputer:
+    """Compute the Brier Score.
+
+    Base class used for computing the Brier Score metric.
+
+    Parameters
+    ----------
+    y_train : np.array, dictionnary or dataframe
+        The target, consisting in the 'event' and 'duration' columns.
+        This is used to fit the IPCW estimator.
+
+    event_of_interest : int or "any", default="any"
+        The event to consider in competitive events setting.
+        "any" indicates that all events except the censoring 0 are
+        considered as a single event.
+        In single event settings, "any" and 1 are equivalent.
+
+    """
+
+    def __init__(
+        self,
+        y_train,
+        event_of_interest="any",
+    ):
+        self.y_train = y_train
+        self.event_train, self.duration_train = check_y_survival(y_train)
+        self.event_ids_ = np.unique(self.event_train)
+        self.any_event_train = self.event_train > 0
+        self.event_of_interest = event_of_interest
+
+        # Estimate the censoring distribution from the training set
+        # using Kaplan-Meier.
+        self.ipcw_est = IpcwEstimator().fit(
+            dict(
+                event=self.any_event_train,
+                duration=self.duration_train,
+            )
+        )
+
+        # Precompute the censoring probabilities at the time of the events on the
+        # training set:
+        self.ipcw_train = self.ipcw_est.predict(self.duration_train)
+
+    def brier_score(self, y_true, y_pred, times):
```

Comment:
I would name this `brier_score_survival` to make it explicit that we expect `y_pred` to hold `1 - incidence_probability`.